src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -305,6 +307,57 @@ public static AssemblyResultSet assembleReads(final AssemblyRegion region,
+                                           final SmithWatermanAligner aligner, final Haplotype refHaplotype, final AssemblyResultSet assemblyResultSet) {
+        final int activeRegionStart = refHaplotype.getAlignmentStartHapwrtRef();
+        final Map<Integer, VariantContext> assembledVariants = assemblyResultSet.getVariationEvents(maxMnpDistance).stream()
+                .collect(Collectors.groupingBy(VariantContext::getStart, Collectors.collectingAndThen(Collectors.toList(), AssemblyBasedCallerUtils::makeMergedVariantContext)));
+
+        final List<Haplotype> assembledHaplotypes = assemblyResultSet.getHaplotypeList();
+        for (final VariantContext givenVC : givenAlleles) {
+            final VariantContext assembledVC = assembledVariants.get(givenVC.getStart());
+            final int givenVCRefLength = givenVC.getReference().length();
+            final Allele longerRef = (assembledVC == null || givenVCRefLength > assembledVC.getReference().length()) ? givenVC.getReference() : assembledVC.getReference();
+            final List<Allele> unassembledGivenAlleles;
+            if (assembledVC == null) {
+                unassembledGivenAlleles = givenVC.getAlternateAlleles();
+            } else {
+                // map all alleles to the longest common reference
+                final Set<Allele> assembledAlleleSet = new HashSet<>(longerRef.length() == assembledVC.getReference().length() ? assembledVC.getAlternateAlleles() :
+                        ReferenceConfidenceVariantContextMerger.remapAlleles(assembledVC, longerRef));
+                final Set<Allele> givenAlleleSet = new HashSet<>(longerRef.length() == givenVCRefLength ? givenVC.getAlternateAlleles() :
+                        ReferenceConfidenceVariantContextMerger.remapAlleles(givenVC, longerRef));
+                unassembledGivenAlleles = givenAlleleSet.stream().filter(a -> !assembledAlleleSet.contains(a)).collect(Collectors.toList());
+            }
+
+            // choose the highest-scoring haplotypes along with the reference for building force-calling haplotypes
+            final List<Haplotype> baseHaplotypes = unassembledGivenAlleles.isEmpty() ? Collections.emptyList() : assembledHaplotypes.stream()
+                    .sorted(Comparator.comparingInt((Haplotype hap) -> hap.isReference() ? 1 : 0).thenComparingDouble(hap -> hap.getScore()).reversed())
+                    .limit(NUM_HAPLOTYPES_TO_INJECT_FORCE_CALLING_ALLELES_INTO)
+                    .collect(Collectors.toList());
+
+            for (final Allele givenAllele : unassembledGivenAlleles) {
+                for (final Haplotype baseHaplotype : baseHaplotypes) {
+                    // make sure this allele doesn't collide with a variant on the haplotype
+                    if (baseHaplotype.getEventMap()!= null && baseHaplotype.getEventMap().getVariantContexts().stream().anyMatch(vc -> vc.overlaps(givenVC))) {
```

Comment:
So if the GGA allele is a SNP that is spanned by a deletion in the discovered variants the it's only added to the reference haplotype, right?  And it will still get output in the vcf?

Reply:
Yes and yes.